corehq/apps/zapier/tests.py

```diff
@@ -435,80 +435,85 @@ def test_create_case(self):
                                data=json.dumps(data),
-                               content_type='application/json')
+                               content_type='application/json',
+                               )
 
-        status = ZapierUpdateCase().post(request)
-        self.assertEqual(status.status_code, 200)
+        self.assertEqual(response.status_code, 200)
         case = self.accessor.get_case(case_id[0])
         self.assertEqual('15', case.get_case_property('price'))
 
     def test_update_case_does_not_exist(self):
-        factory = RequestFactory()
+        client = Client()
         data = {'case_name': 'test1', 'price': '15', 'case_id': 'fake_id'}
-        request = factory.post(self.test_url,
+        response = client.post(reverse(ZapierUpdateCase.urlname, kwargs={'domain': self.domain}) + self.query_string,
                                data=json.dumps(data),
-                               content_type='application/json')
+                               content_type='application/json',
+                               )
 
-        status = ZapierUpdateCase().post(request)
-        self.assertEqual(status.status_code, 403)
+        self.assertEqual(response.status_code, 403)
 
     def test_update_case_wrong_domain(self):
-        factory = RequestFactory()
-        request = factory.post(self.test_url,
+        client = Client()
+        response = client.post(reverse(ZapierCreateCase.urlname, kwargs={'domain': self.domain}) + self.query_string,
                                data=json.dumps(self.data),
-                               content_type='application/json')
-        ZapierCreateCase().post(request)
+                               content_type='application/json',
+                               )
+
+        self.assertEqual(response.status_code, 200)
         case_id = self.accessor.get_case_ids_in_domain()
 
-        factory = RequestFactory()
         data = {'case_name': 'test1', 'price': '15', 'case_id': case_id[0]}
-        request = factory.post("http://commcarehq.org/?domain=me&case_type=watermelon&user_id=test_user&user=test",
+        query_string = "?domain=me&case_type=watermelon&user_id=test_user&user=test"
+        response = client.post(reverse(ZapierUpdateCase.urlname, kwargs={'domain': self.domain}) + query_string,
                                data=json.dumps(data),
-                               content_type='application/json')
+                               content_type='application/json',
+                               )
 
-        status = ZapierUpdateCase().post(request)
-        self.assertEqual(status.status_code, 403)
+        self.assertEqual(response.status_code, 403)
 
     def test_update_case_wrong_type(self):
-        factory = RequestFactory()
-        request = factory.post(self.test_url,
+        client = Client()
+        response = client.post(reverse(ZapierCreateCase.urlname, kwargs={'domain': self.domain}) + self.query_string,
```

Comment:
nit: Since this is duplicated in all of these test methods, you could pull it out into an instance method:
```py
def _post(self, query_string):
    client = Client()
    return client.post(...)
```